### PR TITLE
ASoC: SOF: mediatek: Add mediatek debug dump

### PR DESCRIPTION
**Pull request by yaochunhung**
Hi, 
This PR include two patches :
One add common dedug dump function for mediatek platforms.
Another is for mt8195 debug dump implementation for mt8195 specific debug registers and also call common debug dump.
Please help to review it. Thanks.

### CODE REVIEW

**Comment by yaochunhung** [comment]
SOFCI TEST

File: sound/soc/sof/mediatek/mtk-adsp-common.c [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+				   u32 *stack, size_t stack_words)
+{
+	u32 offset = sdev->dsp_oops_offset;
+
+	/* first read registers */
+	sof_mailbox_read(sdev, offset, xoops, sizeof(*xoops));
+
+	/* then get panic info */
+	if (xoops->arch_hdr.totalsize > EXCEPT_MAX_HDR_SIZE) {
+		dev_err(sdev->dev, "invalid header size 0x%x. FW oops is bogus\n",
```
Comment (ujfalusi) [inline sound/soc/sof/mediatek/mtk-adsp-common.c]: I would drop the `FW oops is bogus` part.

**Review by ujfalusi — COMMENTED** [review]
@yaochunhung, you might want to have dynamic debug level used for the prints, similarly as it is done via `hda_dsp_dump()` or you are happy with the constant dev_info + dev_err combination?

File: sound/soc/sof/mediatek/mt8195/mt8195.c [hunk 2]
```diff
@@ -356,6 +357,31 @@ static int mt8195_get_bar_index(struct snd_sof_dev *sdev, u32 type)
...
+	dbg_inst = snd_sof_dsp_read(sdev, DSP_REG_BAR, DSP_PDEBUGINST);
+	dbg_ls0stat = snd_sof_dsp_read(sdev, DSP_REG_BAR, DSP_PDEBUGLS0STAT);
+	dbg_ls1stat = snd_sof_dsp_read(sdev, DSP_REG_BAR, DSP_PDEBUGLS1STAT);
+	faultbus = snd_sof_dsp_read(sdev, DSP_REG_BAR, DSP_PFAULTBUS);
+	faultinfo = snd_sof_dsp_read(sdev, DSP_REG_BAR, DSP_PFAULTINFO);
+	swrest = snd_sof_dsp_read(sdev, DSP_REG_BAR, DSP_RESET_SW);
+
+	dev_info(sdev->dev, "adsp dump : pc 0x%x, data 0x%x, bus0 0x%x, bus1 0x%x, swrest 0x%x",
+		 dbg_pc, dbg_data, dbg_bus0, dbg_bus1, swrest);
+	dev_info(sdev->dev, "ls0stat 0x%x, ls1stat 0x%x, faultbus 0x%x, faultinfo 0x%x",
```
Comment (plbossart) [inline sound/soc/sof/mediatek/mt8195/mt8195.c]: use %#x

Comment (yaochunhung) [reply sound/soc/sof/mediatek/mt8195/mt8195.c]: I will modify it. thanks.

File: sound/soc/sof/mediatek/mtk-adsp-common.c (hunk 1 above)
Comment (yaochunhung) [reply sound/soc/sof/mediatek/mtk-adsp-common.c]: I will remove it. thanks.

Comment (yaochunhung) [reply sound/soc/sof/mediatek/mtk-adsp-common.c]: > @yaochunhung, you might want to have dynamic debug level used for the prints, similarly as it is done via `hda_dsp_dump()` or you are happy with the constant dev_info + dev_err combination?

@ujfalusi  Thanks for this suggestion. I will check about the dynamic debug level. Thanks.

**Review by plbossart — DISMISSED** [review]

**Review by plbossart — APPROVED** [review]
LGTM

**Review by ranj063 — APPROVED** [review]

**Pull request merged**